### PR TITLE
added validation for invalid id

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [ ] Acceptance tests

### CODE REVIEW

**Comment by AnikaAgiwal2711** [comment]
run e2e test TestAccReplicationReportsDataSource

**Comment by AnikaAgiwal2711** [comment]
run e2e test ReplicationReportsDataSource

File: powerscale/provider/synciq_replication_reports_datasource.go [hunk 1]
```diff
@@ -865,6 +865,10 @@ func (d *ReplicationReportDataSource) Read(ctx context.Context, req datasource.R
 		}
 		rr = append(rr, entity)
 	}
```
Comment (Krunal-Thakkar) [inline powerscale/provider/synciq_replication_reports_datasource.go:867]: Update a logic as discussed

**Review by Krunal-Thakkar — APPROVED** [review]

**Comment by AnikaAgiwal2711** [comment]
run e2e test ReplicationReportsDataSource

**Pull request merged**